### PR TITLE
Added: Editable entity combo box can now use default selected and hav…

### PR DESCRIPTION
**Pull request by dvdvideo1234**
…e icons

### CODE REVIEW

File: garrysmod/lua/vgui/prop_combo.lua [hunk 1]
```diff
@@ -10,13 +10,15 @@ function PANEL:Setup( vars )
 
 	self:Clear()
 
+	local bicon = istable( vars.icons )
```
Comment (GitSparTV) [inline garrysmod/lua/vgui/prop_combo.lua:13]: Why should it be there?

It's used only in loop

Comment (dvdvideo1234) [reply garrysmod/lua/vgui/prop_combo.lua:13]: I put it outside so it shold not be cheched every iteration to generate the boolean control. It is for performance.

Comment (GitSparTV) [reply garrysmod/lua/vgui/prop_combo.lua:13]: The question is why did you moved even further in other commit?

Comment (dvdvideo1234) [reply garrysmod/lua/vgui/prop_combo.lua:13]: Put it with the other declarations. Wherw do you perer it to be moved?

File: garrysmod/lua/vgui/prop_combo.lua [hunk 2]
```diff
@@ -10,13 +10,15 @@ function PANEL:Setup( vars )
 
 	self:Clear()
 
+	local bico = istable( vars.icons )
```
Comment (robotboy655) [inline garrysmod/lua/vgui/prop_combo.lua:13]: This should be moved to just above the loop.

```
	local bico = istable( vars.icons )
	for id, thing in pairs( vars.values or {} ) do
```

should also probably have a better name like `hasIcons`.

**Review by robotboy655 — CHANGES_REQUESTED** [review]
```lua
local Panel = vgui.Create( "DFrame" )
Panel:SetSize( 500, 500 )
Panel:MakePopup()

local DP = vgui.Create( "DProperties", Panel )
DP:Dock( FILL )

local choice = DP:CreateRow( "Choices", "Combo #1: Default" )
choice:Setup( "Combo", {} )
choice:AddChoice( "Allow", true )
choice:AddChoice( "Disallow", false )

local choice = DP:CreateRow( "Choices", "Combo #2: Custom default text" )

local vals = {}

vals[ "Table" ] = {}
vals[ "Function" ] = function() end
vals[ "String" ] = "Hello world"

local ics = {}
ics[ "Table" ] = "icon16/cross.png"

choice:Setup( "Combo", { text = "Select type...", values = vals, icons = ics } )
choice.DataChanged = function( self, data )
    print( "You selected: ", data )
end
```

```
[ERROR] lua/vgui/dimage.lua:99: bad argument #1 to 'Material' (string expected, got table)
  1. Material - [C]:-1
   2. SetImage - lua/vgui/dimage.lua:99
    3. SetIcon - lua/vgui/dbutton.lua:47
     4. OpenMenu - lua/vgui/dcombobox.lua:205
      5. DoClick - lua/vgui/dcombobox.lua:272
       6. unknown - lua/vgui/dlabel.lua:237
```

Not acceptable.

File: garrysmod/lua/vgui/prop_combo.lua [hunk 3]
```diff
@@ -15,8 +15,15 @@ function PANEL:Setup( vars )
 	combo:DockMargin( 0, 1, 2, 2 )
 	combo:SetValue( vars.text or "Select..." )
 
+	local hasIcons, icon = istable( vars.icons )
 	for id, thing in pairs( vars.values or {} ) do
-		combo:AddChoice( id, thing )
+		if(hasIcons) then
```
Comment (FlorianLeChat) [inline garrysmod/lua/vgui/prop_combo.lua:20]: Spacing is required here

Comment (dvdvideo1234) [reply garrysmod/lua/vgui/prop_combo.lua:21]: Fixed in fbf1e00debbf39c8e81097dc351453f2ff98d30d

Comment (Kefta) [reply garrysmod/lua/vgui/prop_combo.lua:21]: You did not fix it, there is still no space between if and the parenthesis.

**Review by Kefta — CHANGES_REQUESTED** [review]

File: garrysmod/lua/vgui/prop_combo.lua (hunk 3 above)
Comment (Kefta) [reply garrysmod/lua/vgui/prop_combo.lua]: Still missing the space on the other side of the if..

**Pull request merged**